types/types.go

```diff
@@ -0,0 +1,200 @@
+package types
+
+//go:generate go run ./gen/gen.go
+
+import (
+	"fmt"
+	"github.com/filecoin-project/go-address"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+type IPCAddress struct {
+	SubnetID   SubnetID
+	RawAddress address.Address
+}
+
+func (i IPCAddress) String() string {
+	// `-` is used as delimiter instead of `/`
+	// `/` is harder to parse as `SubnetId` contains `/`, which makes it difficult to
+	// determined which is the start of Address
+	return strings.Join([]string{i.SubnetID.String(), i.RawAddress.String()}, IPCAddrSeparator)
+}
+
+func IPCAddressFromString(addr string) (*IPCAddress, error) {
+	r := strings.Split(addr, "-")
+	if len(r) != 2 {
+		return nil, fmt.Errorf("invalid IPCAddress string type") // TODO Create new error, TODO Define Undef for IPCAddress
+	}
+	rawAddress, err := address.NewFromString(r[1])
+	if err != nil {
+		return nil, err
+	}
+	ptrSubnetID, err := SubnetIDFromString(r[0])
+	if err != nil {
+		return nil, err
+	}
+	return &IPCAddress{*ptrSubnetID, rawAddress}, nil
+}
+
+type SubnetID struct {
+	Parent string
+	Actor  address.Address
+}
+
+var id0, _ = address.NewIDAddress(0) //TODO check with Alfonso
+
+const (
+	RootStr          = "/root"
+	SubnetSeparator  = "/"
+	UndefStr         = ""
+	IPCAddrSeparator = "-"
```

Comment:
the separator is `:` not `-`. https://github.com/consensus-shipyard/ipc-actors/blob/877177fa3f56ead824da53084b9957619e823d59/sdk/src/address.rs#L9